### PR TITLE
chore: enable auto-update (with updatecli) of git, git-lfs, docker-compose and maven

### PR DESCRIPTION
**Pull request by dduportal**
This PR introduces an automatic update (ala dependabot) of some dependencies of this repository.

It uses updatecli, as our usual tool to track the versions of the following elements:

- git
- git-lfs
- maven
- docker-compose
- jdk8 (adoptium)
- jdk11 (adoptium)

This PR updates the formatting of the file `.auto.pkrvars.hcl` as updatecli would mess up the formatting.

This PR adds a Github action to ensure that there is at least a daily execution (as ci.jenkins.io should not have any credential allowing write on repositories).
The `updatecli diff` command is expected to be run on EACH commit/PR/branch to ensure that a given change does not break updatecli without the contributor knowing it.

To validate this PR, once merged, we should have a PR opened for the latest Maven 3.8.2 release.

### CODE REVIEW

File: updatecli/updatecli.d/compose.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    # Version is retrieved from the GitHub source mirror: if there is a release, then you're sure that the artifacts are available
+    kind: githubRelease
+    name: Get the latest Docker Compose version
+    spec:
+      owner: "docker"
+      repository: "compose"
+      token: "{{ requiredEnv .github.token }}"
+      username: "{{ .github.username }}"
+      versionFilter:
+        kind: regex
```
Comment (olblak) [inline updatecli/updatecli.d/compose.yml:14]: It seems that https://github.com/docker/compose uses semantic versioning so something like seems easier to read

```suggestion
        kind: semver
        pattern: "~1"
```

Comment (olblak) [inline updatecli/updatecli.d/compose.yml:14]:
```suggestion
        kind: semver
        pattern: "~1"
```

File: updatecli/updatecli.d/compose.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        pattern: "^(\\d*).(\\d*).(\\d*)$"
```
Comment (olblak) [inline updatecli/updatecli.d/compose.yml:15]:
```suggestion
```

File: updatecli/updatecli.d/git-lfs.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    # Version is retrieved from the GitHub source mirror: if there is a release, then you're sure that the artifacts are available
+    kind: githubRelease
+    name: Get the latest Git-LFS version
+    spec:
+      owner: "git-lfs"
+      repository: "git-lfs"
+      token: "{{ requiredEnv .github.token }}"
+      username: "{{ .github.username }}"
+      versionFilter:
+        kind: regex
```
Comment (olblak) [inline updatecli/updatecli.d/git-lfs.yml:14]: Same comment here, please be informed of this bug https://github.com/updatecli/updatecli/issues/244
the semver automatically remove the prefix "v" just before fetching changelogs

```suggestion
        kind: semver
```

File: updatecli/updatecli.d/git-lfs.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        pattern: "^v(\\d*).(\\d*).(\\d*)$"
```
Comment (olblak) [inline updatecli/updatecli.d/git-lfs.yml:15]:
```suggestion
```

File: updatecli/updatecli.d/git-lfs.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    transformers:
```
Comment (olblak) [inline updatecli/updatecli.d/git-lfs.yml:16]:
```suggestion
```

File: updatecli/updatecli.d/git-lfs.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      - trimPrefix: "v"
```
Comment (olblak) [inline updatecli/updatecli.d/git-lfs.yml:17]: Character 'v' is already remove in the semantic versioning
```suggestion
```

File: updatecli/updatecli.d/git.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    # Version is retrieved from the GitHub source mirror: if there is a release, then you're sure that the artifacts are available
+    kind: githubRelease
+    name: Get the latest Git version
+    spec:
+      owner: "git"
+      repository: "git"
+      token: "{{ requiredEnv .github.token }}"
+      username: "{{ .github.username }}"
+      versionFilter:
+        kind: regex
```
Comment (olblak) [inline updatecli/updatecli.d/git.yml:14]:
```suggestion
        kind: semver
```

File: updatecli/updatecli.d/git.yml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        pattern: "^v(\\d*).(\\d*).(\\d*)$"
```
Comment (olblak) [inline updatecli/updatecli.d/git.yml:15]:
```suggestion
```

File: updatecli/updatecli.d/git.yml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    transformers:
```
Comment (olblak) [inline updatecli/updatecli.d/git.yml:16]:
```suggestion
```

File: updatecli/updatecli.d/git.yml [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+      - trimPrefix: "v"
```
Comment (olblak) [inline updatecli/updatecli.d/git.yml:17]:
```suggestion
```

**Review by olblak — CHANGES_REQUESTED** [review]
Few minor suggestion which would  avoid relying on regex :D

**Comment by dduportal** [comment]
- Taken @olblak review feedbacks
- Added jdk8 and jdk11 (as #78 is merged)
- Rebased from main branch

**Comment by dduportal** [comment]
- Rebased on latest main branch (with Jenkins job migrated to infra.ci)
- Applied @timja proposal from #88 to enable release-drafter (paving the way for automatic release)
- Increased timeout and VM size for the Windows-2019 Azure VM as it seems sometime (until we find the real blocking issue that requires connecting to the instance, read the sysprep log and see what is going wrong)

**Review by timja — APPROVED** [review]

**Comment by dduportal** [comment]
The Windows build on Azure is still blocked: since the rest passed successfully, I'm merging this

**Pull request merged**